python/lsst/cp/pipe/defects.py

```diff
@@ -246,7 +247,15 @@ def findHotAndColdPixels(self, exp, nSigma):
 
                 threshold = afwDetection.createThreshold(nSig, 'stdev', polarity=polarity)
 
-                footprintSet = afwDetection.FootprintSet(ampImg, threshold)
+                try:
+                    footprintSet = afwDetection.FootprintSet(ampImg, threshold)
+                except InvalidParameterError:
+                    # This occurs if the image sigma value is 0.0.
```

Comment:
Is there any other way this could happen?  Testing exceptions should be the last resort.  Can't the `threshold` value be checked?

Reply:
The `threshold` itself is fine.  It's the image sigma value coming out as zero that causes the `InvalidParameterError`.  Manually calculating the image statistics seemed to be less efficient than doing the try/except, as most amplifiers should work fine.